java-checks/src/main/java/org/sonar/java/checks/CatchOfThrowableOrErrorCheck.java

```diff
@@ -111,7 +111,10 @@ public void visitThrowStatement(ThrowStatementTree tree) {
     private boolean isGuavaCloserRethrow(ExpressionTree expression) {
       if (expression.is(Tree.Kind.METHOD_INVOCATION)) {
         MethodInvocationTree mit = (MethodInvocationTree) expression;
-        if (MATCHERS.matches(mit)) {
+        // In case of broken semantics, the method matcher cannot identify the method invocation.
+        // To avoid raising FPs in these cases, we accept is as is and look at the first argument
+        if (MATCHERS.matches(mit) ||
```

Comment:
To me, this is the wrong fix. if the invocation is unknown, you can not assume it's equivalent to guava's closer, especially relying only on number of arguments! Without semantic, some issues should simply not be raised, because when methods are unknown, just reporting nothing is way safer. We don't assume anything.

Also, if we don't know what is in the thrown from body of the `try { }`, then reporting an issue on any `catch` is wrong. and I don't see any tests of this.
```java
try {
  unknown();
} catch (Throwable e) { } // should not raise any issue - the user might not have the choice and we don't know it!
```